pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -106,7 +107,7 @@ func NewGuardController(
 	return factory.New().WithInformers(
 		kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
 		kubeInformersClusterScoped.Core().V1().Nodes().Informer(),
-	).WithSync(c.sync).WithSyncDegradedOnError(operatorClient).ToController("GuardController", eventRecorder), nil
+	).ResyncEvery(20*time.Second).WithoutEventHandlers().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).ToController("GuardController", eventRecorder), nil
```

Comment:
Have you considered using `ResyncEvery()` without `WithInformers()` instead?

If you need `WithInformers()` just for the cache-waiting behavior, would a new `WithCachesSynced(informers)` be a more intuitive API rather than `WithoutEventHandlers()`?